Components/Sensors/BME280.cs

```diff
@@ -0,0 +1,367 @@
+    ///     NOTE: This is for a similar device, the BMP280 (P for pressure only). This describes register LSB/MSB order, but has no info about humidity.
+    /// </summary>
+    public class BME280 : ISensor
+    {
+        public string System { get; set; }
+
+        public static readonly Config DefaultConfig = new Config()
+        {
+            HumidityOversampling = Oversampling.OS_1x,
+            MeasureHumidity = true,
+            PressureOversampling = Oversampling.OS_1x,
+            MeasurePressure = true,
+            TemperatureOversampling = Oversampling.OS_1x,
+
+            Mode = Mode.NORMAL,
+            StandbyTime = StandbyTime.TIME_500us,
+            IIRFilterTimeConstant = FilterCoefficient.FILTER_OFF,
+
+            Use3WireSPI = false
+        };
+
+        public double Temperature { get; private set; }
+        public double Humidity { get; private set; }
+        public double Pressure { get; private set; }
+
+        private readonly II2CBus I2CBus;
+        private readonly byte I2CAddress;
+        private readonly ISPIBus SPIBus;
+        private readonly IDigitalOut SPICS;
+        private readonly bool IsSPI;
+
+        private CompensationParameters CompParams;
+        private Config Configuration;
+
+        /// <summary> Initializes a <c>BME280</c> environmental sensor, using the I2C protocol (the device supports SPI as well). </summary>
+        /// <param name="I2CBus"> The I2C bus used to communicate with the device. </param>
+        /// <param name="DeviceAddress"> The I2C address of the device, 0x76 or 0x77 usually. </param>
+        public BME280(II2CBus I2CBus, byte DeviceAddress = 0x76)
+        {
+            this.IsSPI = false;
+            this.I2CBus = I2CBus;
+            this.I2CAddress = DeviceAddress;
+        }
+
+        /// <summary> WARNING: <c>BME280</c> has not been tested with SPI. </summary>
+        /// <param name="SPIBus"> The SPI bus used to communicate with the device. </param>
+        /// <param name="ChipSelect"> The output to use to signal chip select to the device. </param>
+        public BME280(ISPIBus SPIBus, IDigitalOut ChipSelect)
+        {
+            this.IsSPI = true;
+            this.SPIBus = SPIBus;
+            this.SPICS = ChipSelect;
+        }
+
+        /// <summary> Sets the device's registers to match the given <c>Configuration</c>. </summary>
+        /// <param name="Configuration"> The configuration to apply. </param>
+        public void Configure(Config Configuration)
+        {
+            this.Configuration = Configuration;
+            this.CompParams = ReadCompVals();
+            ChangeMode(Mode.SLEEP);
+
+            byte[] RawData = Read((byte)Register.CTRL_HUM, 4);
+
+            byte osrs_h = (byte)(Configuration.MeasureHumidity ? (byte)Configuration.HumidityOversampling : 0);
+            byte Ctrl_Hum = (byte)((RawData[0] & 0b1111_1000) | osrs_h & 0b111);
```

Comment:
Not a big deal at all, but it stood out to me that `Ctrl_Hum` is capitalized and the other bytes here aren't. Is there a reason?

Reply:
The `osrs_*` bytes aren't capitalized because that's what that data is called, and I'm honestly noit sure what it stands for, so I couldn't capitalize it properly. (i.e. should it be OsRS? OSRS? OsrS? etc). So I capitalized the ones where I knew what it was, and left the ones I didn't know as is.